doc/developer_guide/index.md

```diff
@@ -158,22 +187,32 @@ pixi run test-ui
 The documentation can be built with the command:
 
 ```bash
-pixi run docs-build
+pixi run build-docs
 ```
 
-As Panel uses notebooks for much of the documentation, this will take significant time to run (around an hour).
+As Panel uses notebooks for much of the documentation, this will take significant time to run (around an hour) the first time you run the command.
 
-A development version of Panel can be found [here](https://holoviz-dev.github.io/panel/). You can ask a maintainer if they want to make a dev release for your PR, but there is no guarantee they will say yes.
+To speed up your workflow, you may run specific sub tasks via the commands:
 
-To be able to run cells interactively you need `pyodide` server, this can be ran with:
+```bash
+pixi run _docs-build-api-reference
```

Comment:
I've never understood the docs build flow before. But now I think I do. Making the sub tasks more visible and giving them more readable names will help a me lot. I believe it will help others too.

Reply:
For example all the broken links we have. Now I think I now better how to fix it because I understand how the `nbsite` build will see the examples.

Reply:
> Making the sub tasks more visible ... I believe it will help others too.

I explicitly made them non-visible. The biggest problem is that one of the tasks takes 95 % or more of the time; this should be improved at a fundamental level with how `nbsite` builds the documentation. 

Two things should be considered when working with development tools: using them and debugging/fixing problems with them.

Most contributors should "just" use the development tool, which should be as easy as possible. This is why it is one command and hides the subcommands. Most users should not need to remember (or even know) five subcommands and which order they should be run in.

You are having problems debugging or fixing a problem caused by the development tool you are using, which is great! But it is, or should be, limited to a handful of people. 

This suggestion falls into the latter category and could, in my opinion, confuse new developers more than help them.

> ... and giving them more readable names will help a me lot.

No problem with better names.

Reply:
But there is no need to spend time on creating api Gallery or Pyodide  output if I'm working on some markdown document.

Reply:
Agree with that, that said only certain subtasks should be pulled out, I'd say currently this makes sense:

- `docs-build-sphinx`
- `docs-build-pyodide-gallery`

and then I'd ideally to have a `docs-preview` task so you can preview individual docs pages without building the entire sphinx site.